### DISCUSSION TITLE
xsh script needed for neovim switcher

### DISCUSSION THREAD

**Discussion by pannet1** [xonsh/xonsh#5112] [General]
though i am using python for many of my work, solicit someone's help to provide me with a xsh script for switching neovim configuration

[neovim switcher user contributed scripts for various shells](https://gist.github.com/elijahmanor/b279553c0132bfad7eae23e34ceb593b)

**Comment by pannet1** [comment]
i lacked the confidence to do this myself. just adding the right alias solved the purpose. Even better i updated the neovim switcher gists for how to do it on XONSH

[xonsh neovim switcher](https://gist.github.com/elijahmanor/b279553c0132bfad7eae23e34ceb593b?permalink_comment_id=4526616#gistcomment-4526616)

hope this is useful for someone. my first smallest contribution to xonsh

**Reply by anki-code** [reply]
Thanks and welcome! Mark your comment as an answer.

**Reply by pannet1** [reply]
Hi, @anki-code Mark as answer is unavailable probably because i started the discussion.

**Comment by slacksystem** [comment]
If you didn't want to modify the environment with your alias, i.e. leaving `$NVIM_APPNAME` changed, you could do 
```xsh
@aliases.register
def _nvim_astro(args):
    $NVIM_APPNAME='AstroVim' nvim @(args)
nvim-astro
```
This will revert `$NVIM_APPNAME` to whatever it was before running the command after it runs nvim, so the next time you run just `nvim`, it wouldn't still be running astronvim
